File: project/backend/src/main/java/dev/lueem/model/Article.java
package dev.lueem.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.Getter;
import java.math.BigDecimal;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Article {


    @JsonProperty("name")
    private String name;

    @JsonProperty("price")
    private BigDecimal price;

    @JsonProperty("quantity")
    private int quantity;

    @JsonProperty("discount")
    private BigDecimal discount;

    @JsonProperty("total")
    private BigDecimal total;

    @JsonProperty("category")
    private String category;

    // optional?
    @JsonProperty("id")
    private Long id;

}

File: project/backend/src/main/kotlin/dev/lueem/model/Article.kt
package dev.lueem.model

import com.fasterxml.jackson.annotation.JsonProperty
import java.math.BigDecimal

data class Article(
    @JsonProperty("name")
    var name: String? = null,

    @JsonProperty("price")
    var price: BigDecimal? = null,

    @JsonProperty("quantity")
    var quantity: BigDecimal? = null,

    @JsonProperty("discount")
    var discount: BigDecimal? = null,

    @JsonProperty("total")
    var total: BigDecimal? = null,

    @JsonProperty("category")
    var category: String? = null,

    @JsonProperty("id")
    var id: Long? = null
)
